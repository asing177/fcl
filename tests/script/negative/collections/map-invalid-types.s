
global map<account, int> shares = ();

transition initial -> terminal;

@initial
insertTest(account a, datetime d, decimal<6> amount) {
  mapInsert(d, 5, shares);
  mapinsert(a, amount, shares);
  mapinsert(a, 500, shares);
  terminate();
}

@initial
deleteTest(asset<decimal<5>> a) {
  mapDelete(a, shares);
  terminate();
}

@initial
lookupTest(msg m) {
  v = lookup(m,shares);
  terminate();
}
