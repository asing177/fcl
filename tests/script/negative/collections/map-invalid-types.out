@initial
lookupTest(msg m) {
-----------^
  v = lookup(m,shares);
  terminate();

"<stdin>" (line 21, column 12):
unexpected "m"
expecting comma seperated list or ")"